main.cpp

```diff
@@ -636,13 +589,36 @@ int main(int argc, char *argv[])
     parser->insertOption(17,"--http-proxy 1");
     parser->insertOption(18,"--socks5-proxy 1");
 
+    parser->insertOption(19,"--no-stats"); //don't collect and send anonymous statistics
+
 
     //...............
 
     parser->parse(opts);
 
+    if(!parser->isParamExist("no-stats")){
+        // Application instance definition
 
-    int ret;
+        QString AAID=getAIID();
+        QString OS=getOS();
+
+        utsname u;
+        uname(&u);
+
+        QString PLATFORM=u.machine;
+        QString DISTR=QString(u.sysname)+" "+QString(u.release);
+
+
+        MSRequest* req=new MSRequest();
```

Comment:
@MasterSoft24  small memory leak is still memory leak. Because it was not found by any tool or testing, I cant be sure that such memory leaks no not exists in deeper code, 

@MasterSoft24 you started to develop a greate idea, but please make the quality checking better. This is required if you wish stable work of you system.  Currently code quality is rather poor. If you with to handle it as pet project for your own, it is OK, but for distributions, adding to repositories, popularization and involving new contributors you rely need to add some best practices of open source projects

I would suggest you this as first reading : 
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines
https://danielkummer.github.io/git-flow-cheatsheet/ 
https://lefticus.gitbooks.io/cpp-best-practices/content/

Briefly you need in you project:
- Unit testing
- Functional testing for releases
- Static analyzer
- Code style and conventions
- Code beautifier
- git flow (with branching policy and review) 
- Continuous integration
- Continuous delivery (when you will be ready to add you project to repos)